python/cudf/cudf/pandas/__init__.py

```diff
@@ -50,16 +52,30 @@ def install():
             mr = rmm.mr.CudaAsyncMemoryResource(initial_pool_size=free_memory)
             rmm.mr.set_current_device_resource(mr)
         elif rmm_mode == "managed":
-            mr = rmm.mr.ManagedMemoryResource()
+            mr = rmm.mr.PrefetchResourceAdaptor(rmm.mr.ManagedMemoryResource())
             rmm.mr.set_current_device_resource(mr)
+            enable_prefetching = True
         elif rmm_mode == "managed_pool":
-            mr = rmm.mr.PoolMemoryResource(
-                rmm.mr.ManagedMemoryResource(),
-                initial_pool_size=free_memory,
+            mr = rmm.mr.PrefetchResourceAdaptor(
+                rmm.mr.PoolMemoryResource(
+                    rmm.mr.ManagedMemoryResource(),
+                    initial_pool_size=free_memory,
+                )
             )
             rmm.mr.set_current_device_resource(mr)
+            enable_prefetching = True
         else:
             raise ValueError(f"Unsupported rmm mode: {rmm_mode}")
+        from cudf._lib import pylibcudf
+
+        if enable_prefetching:
+            for item in {
```

Comment:
I think the term for these is a “key.”
```suggestion
            for key in {
```